rest/server_context_test.go

```diff
@@ -163,3 +165,197 @@ func (client *MockClient) RespondToGET(url string, response *http.Response) {
 	tripper := client.Transport.(*mockTripper)
 	tripper.getURLs[url] = response
 }
+
+func TestRemoveDatabase(t *testing.T) {
+	server := "walrus:"
+	bucketName := "imbucket"
+	databaseName := "imdb"
+
+	serverConfig := &ServerConfig{CORS: &CORSConfig{}, AdminInterface: &DefaultAdminInterface}
+	serverContext := NewServerContext(serverConfig)
+	bucketConfig := BucketConfig{Server: &server, Bucket: &bucketName}
+	dbConfig := &DbConfig{BucketConfig: bucketConfig, Name: databaseName, AllowEmptyPassword: true}
+
+	dbContext, err := serverContext.AddDatabaseFromConfig(dbConfig)
+	assert.NoError(t, err, "No error while adding database to server context")
+	assert.Equal(t, server, dbContext.BucketSpec.Server)
+	assert.Equal(t, bucketName, dbContext.BucketSpec.BucketName)
+
+	status := serverContext.RemoveDatabase(databaseName)
+	assert.True(t, status, "Database should be removed from server context")
+	dbContext, err = serverContext.GetDatabase(databaseName)
+	assert.Nil(t, dbContext, "Database context should not be available!")
+}
+
+func TestAllDatabaseNames(t *testing.T) {
+	server := "walrus:"
+	bucketName := "imbucket"
+	var databaseNames []string
+
+	serverConfig := &ServerConfig{CORS: &CORSConfig{}, AdminInterface: &DefaultAdminInterface}
+	serverContext := NewServerContext(serverConfig)
+	bucketConfig := BucketConfig{Server: &server, Bucket: &bucketName}
+
+	for i := 0; i < 5; i++ {
```

Comment:
Reducing this test to creating only two databases seems like a reasonable thing to do. It will allow you to simplify the test by:
- remove the looping setup stuff
- only removing one database out of the list (and removes the need for those string slice helpers)
- assert.Len + assert.Contains + assert.NotContains

Reply:
Makes sense; I've revised the changes in latest commit.